maestro-cli/src/main/java/maestro/cli/util/MaestroFactory.kt

```diff
@@ -24,43 +24,38 @@ import maestro.utils.SocketUtils.isPortOpen
 import dadb.Dadb
 
 object MaestroFactory {
-
     @Suppress("NAME_SHADOWING")
     fun createMaestro(platform: String?, host: String?, port: Int?): Maestro {
         val host = host ?: "localhost"
         val port = port ?: 10882
 
         return when (platform) {
             null -> {
-                try {
-                    connectAndroid(host)
-                } catch (e: Exception) {
-                    connectIos(host, port)
+                val dadb = Dadb.discover(host)
+
+                if (dadb != null) {
+                    if (isPortOpen(host, port)) {
```

Comment:
If user provided their own port for Android device - this feel fail, which is not what we want